rails/app/controllers/admin/themes_controller.rb

```diff
@@ -0,0 +1,27 @@
+require 'pry'
+module Admin
+  class ThemesController < Admin::ApplicationController
+    def create
+      if theme_params[:active]
+        deactivate_prev_theme
+      end
+      super
+    end
+
+    def update
+      if theme_params[:active]
+        deactivate_prev_theme
+      end
+      super
+    end
+    
+    def deactivate_prev_theme
+      prev = Theme.find_by(active: true)
+      prev.update(active: false)
```

Comment:
I think if we change these lines to this one:

```suggestion
      Theme.where(active: true).update_all(active: false)
```

It would stop any errors from trying to run `update` when no record is active found. (`update_all` will still run on an empty relation and commit no updates). It would also make it so if we somehow ended up with multiple themes active, it would ensure ALL of them are marked as inactive.

If we would rather leave it the same, we can do a guard check against no-previously active theme on line 20 to avoid errors:

```suggestion
      prev = Theme.find_by(active: true)
      prev.update(active: false) if prev.present?
```